drivers/docker/util/util_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package util
+
+import (
+	"testing"
+)
+
+func TestCalculateCPUPercent(t *testing.T) {
+	tests := []struct {
+		name      string
+		newSample uint64
+		oldSample uint64
+		newTotal  uint64
+		oldTotal  uint64
+		cores     int
+		expected  float64
+	}{
+		{"valid case", 100, 50, 200, 100, 4, 200.0},
+		{"zero denominator", 100, 50, 100, 100, 4, 0.0},
+		{"negative case", 50, 100, 100, 200, 4, 0.0},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := CalculateCPUPercent(tt.newSample, tt.oldSample, tt.newTotal, tt.oldTotal, tt.cores)
+			if result != tt.expected {
+				t.Errorf("expected %f, got %f", tt.expected, result)
+			}
```

Comment:
Let use [`must.Eq`](https://github.com/shoenig/test/blob/a74f9d9f28e67b84e7d8230a0287e5ed04230172/must/must.go#L104) for this comparison.